### DISCUSSION TITLE
expose active keyboard layout

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#335] [Ideas]
hey,
it would be nice if we can expose the active keyboard layout, such as to e.g. visualize this like with waybar for sway / hyprland.
i don't know if there is a standard way to do this.

**Comment by YaLTeR** [comment]
Do you happen to know how sway and others expose it?

**Reply by KiaraGrouwstra** [reply]
they know thru wlroots' `keymap`. i guess here you'd use smithay's `seat/keyboard` as you used to swap them.
i'm not sure i get in what way/format they expose the info, but fwiw here's how waybar talks to [sway](https://github.com/Alexays/Waybar/blob/master/src/modules/sway/language.cpp) / [hyprland](https://github.com/Alexays/Waybar/blob/master/src/modules/hyprland/language.cpp) on this.

**Reply by YaLTeR** [reply]
Thanks

**Comment by antieksi** [comment]
Any updates on this? I'd humbly propose that viewing current layout could be done through something like `niri msg layout`, similar to workspaces.

**Reply by YaLTeR** [reply]
No updates yet

**Comment by YaLTeR** [comment]
I added it in f19e1711a7999f9b4d97cf1a1b2e03f273b2f120: `niri msg keyboard-layouts`
